tiledb/sm/array_schema/dimension.cc

```diff
@@ -1591,16 +1598,15 @@ Status Dimension::check_tile_extent() const {
     case Datatype::TIME_AS:
       return check_tile_extent<int64_t>();
     default:
-      return LOG_STATUS(Status_DimensionError(
-          "Tile extent check failed; Invalid dimension domain type"));
+      throw DimensionException(
+          name() + " [check_tile_extent] Invalid dimension domain type");
   }
 }
 
 template <class T>
 Status Dimension::check_tile_extent() const {
   if (domain_.empty())
-    return LOG_STATUS(
-        Status_DimensionError("Tile extent check failed; Domain not set"));
+    throw DimensionException(name() + " [check_tile_extent] Domain not set");
```

Comment:
Same comment as above (drop `[check_tile_name]` and make it more explicit what `name()` refers to).